### PR TITLE
Feature/exemplar smoke test

### PR DESCRIPTION
**Pull request by TheSpartan1980**
# Description

This PR introduces the test automation framework for acceptance tests, 3 tests have been written as an example of what we can do in terms of automating the designer. The test are no yet ready to run on CI as the default browser is Chrome and no changes have been made to incorporate them in the CI config file. Documentation will be updated in due course alongside reporting and screenshots on test failure.

- Added webdriverio as a dependency
- A basic feature consisting of 3 tests
- Step definitions and PageObjects written in JS

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested

The tests were run locally against the designer and developed using the red-green refactor cycle. Files were linted using the Prettier extension in VS Code.

# Checklist:

- [x] My changes do not introduce any new linting errors 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,16 @@
...
     "@hapi/code": "^8.0.1",
     "@hapi/lab": "^24.0.0",
+    "@rpii/wdio-html-reporter": "^6.1.1",
+    "@wdio/cli": "^6.6.0",
+    "@wdio/cucumber-framework": "^6.6.0",
+    "@wdio/local-runner": "^6.6.0",
+    "@wdio/mocha-framework": "^6.6.0",
+    "@wdio/selenium-standalone-service": "^6.6.0",
+    "@wdio/spec-reporter": "^6.6.0",
+    "@wdio/sync": "^6.6.0",
```
Comment (jenbutongit) [inline package.json:53]: could you move these into test/package.json please. When you're thinking about running these in CI, we can set up a yarn workspace for `test`

File: test/features/step-definitions/steps.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  addComponentPage.selectComponentByName(componentName);
+  addComponentPage.completeDateField(
+    "Date of Birth",
+    "dateOfBirth",
+    "Please enter your date of birth using the format dd/mm/yyyy"
+  );
+});
+
+Then("the Date field control is displayed in the page", () => {
+  chai.expect(FormDesignerPage.dropdown(this.pageName).isDisplayed()).to.be.true;
```
Comment (jenbutongit) [inline test/features/step-definitions/steps.js:32]: did you mean to use two different expects? you can just do `const { expect } = require("chai")` so you don't have to do `chai.`

File: test/features/step-definitions/steps.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+When("I choose {string} from the designer menu", (menuOption) => {
+  MenuSection.buttonByName(menuOption).click();
+  this.newPageName = "Personal Details";
+  EditPageSection.pageTitle.setValue(this.newPageName);
+  EditPageSection.saveBtn.click();
+});
+
+Then("the page is added in the designer", () => {
+  var pageName = this.newPageName.toLowerCase().replace(" ", "-");
```
Comment (jenbutongit) [inline test/features/step-definitions/steps.js:59]: does prettier let you use `var`? you're safe from the linter... for now.. `/test` shall not be a lawless place

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: test/features/step-definitions/steps.js (hunk 3 above)
Comment (TheSpartan1980) [reply test/features/step-definitions/steps.js:59]: @jenbutongit it didn't complain however I don't particularly like var, would let be more acceptable?

File: test/features/step-definitions/steps.js (hunk 2 above)
Comment (TheSpartan1980) [reply test/features/step-definitions/steps.js:32]: So sneakily expect is built into webdriverio by default and has a bunch of default matchers, however I felt special enough to use chai in certain cases :)

File: package.json (hunk 1 above)
Comment (TheSpartan1980) [reply package.json:53]: I can indeed.

File: test/features/step-definitions/steps.js (hunk 3 above)
Comment (jenbutongit) [reply test/features/step-definitions/steps.js:59]: `const` :p

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**